lib/fs.js

```diff
@@ -41,6 +41,25 @@ const isWindows = process.platform === 'win32';
 const DEBUG = process.env.NODE_DEBUG && /fs/.test(process.env.NODE_DEBUG);
 const errnoException = util._errnoException;
 
+function getOptions(options, defaultOptions) {
+  if (options === null || options === undefined ||
+      typeof options === 'function') {
+    return defaultOptions;
+  }
+
+  if (typeof options === 'string') {
+    defaultOptions.encoding = options;
```

Comment:
I was thinking this case: someone submits a pull-request which shares `defaultOptions` in some `getOptions()` calls, one of them reaches here, it'd cause the `defaultOptions` are not expected, too.

I know this is internal objects, but here avoiding this way will decrease the bug possibility, why not do this in that way? Just a suggestion from my consideration, totally LGTM :-)